stripe/stripe-android
# [Connect SDK] Add camera support to the Connect SDK

# Summary

Hooks the webview's camera permission into the Android app's camera permission so that the user is shown a permissions modal when the connect SDK requests a camera permission.

[MXMOBILE-2913](https://jira.corp.stripe.com/browse/MXMOBILE-2913)

# Motivation
Certain embedded components (particularly any that use Identity) make use of the camera, and the SDK needs to properly handle camera permissions to make sure these features work.

# Testing

The easiest way I found to test was:
1. Go to payouts
2. Select "Payout" button
3. Select "Add account" under the dropdown of Stripe accounts to select
4. Select an existing account and click the "...", then "edit"
5. Click "you can verify your identity instead"
6. This opens the Identity screen, at which point you can go through the identity flow with the camera being used to scan an ID (in test mode select "Proceed" under "Preview user experience" to make sure you get the camera pop-up)

# Screenshots

### Granting permission


https://github.com/user-attachments/assets/f7e2f88d-ebbb-4ecb-bcff-445ec7c307a0



### Rejecting permission


https://github.com/user-attachments/assets/08c0dac4-bb7c-41ac-81c3-1487c0496f66
